### PR TITLE
vm: trivial bugs fixes

### LINKED ISSUE #3195: --license-type in az vm create?
### Description
We would like to deploy Windows servers with the Hybrid Use Benefit.  To do this with 1.0 we could specify `--license-type Windows_Server`, but it seems that 2.0 does not support this flag.  Is this intentional or is there a way to achieve this using 2.0?

---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: macOS CLI (tried docker as well)

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here: azure-cli (2.0.4)

**OS Version:** What OS and version are you using?  
Answer here: mac 10.12.4

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: zsh

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #3195 
1. Expose license type for windows VM
2. Expose storage account resource id for disk/snapshot creation (only when the source is blob)

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -234,6 +234,8 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
     register_cli_argument(scope, 'os_caching', options_list=['--storage-caching', '--os-disk-caching'], arg_group='Storage', help='Storage caching type for the VM OS disk.', **enum_choice_list([CachingTypes.read_only.value, CachingTypes.read_write.value]))
     register_cli_argument(scope, 'data_caching', options_list=['--data-disk-caching'], arg_group='Storage', help='Storage caching type for the VM data disk(s).', **enum_choice_list(CachingTypes))
 
+    register_cli_argument(scope, 'license_type', help='the license type if the Windows server image or disk being used was licensed on-premises')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: This looks like it should be an enum but the help text reads like a flag.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -284,10 +286,11 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
 
 for scope in ['disk', 'snapshot']:
     register_extra_cli_argument(scope + ' create', 'source', validator=process_disk_or_snapshot_create_namespace,
-                                help='source to create the disk from, including a sas uri to a blob, managed disk id or name, or snapshot id or name')
+                                help='source to create the disk from, including a sas blob uri to a blob, managed disk id or name, or snapshot id or name')
     register_cli_argument(scope, 'source_blob_uri', ignore_type)
     register_cli_argument(scope, 'source_disk', ignore_type)
     register_cli_argument(scope, 'source_snapshot', ignore_type)
+    register_cli_argument(scope, 'source_storage_account_id', help='used when source blob is in a different subscription')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Why is this necessary?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 3]
```diff
@@ -748,12 +748,14 @@ def process_vm_create_namespace(namespace):
     _validate_vm_create_auth(namespace)
     if namespace.secrets:
         _validate_secrets(namespace.secrets, namespace.os_type)
-
+    if namespace.license_type and namespace.os_type.lower() != 'windows':
+        raise CLIError('usage error: --license-type is only applicable on windows vm')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: nit: Windows VM

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 4]
```diff
@@ -857,6 +859,8 @@ def process_disk_or_snapshot_create_namespace(namespace):
     if namespace.source:
         try:
             namespace.source_blob_uri, namespace.source_disk, namespace.source_snapshot = _figure_out_storage_source(namespace.resource_group_name, namespace.source)  # pylint: disable=line-too-long
+            if not namespace.source_blob_uri and namespace.source_storage_account_id:
+                raise CLIError('usage error: --source_storage_account_id is only used with blob uri')
         except CloudError:
             raise CLIError("Incorrect '--source' usage: --source VHD_BLOB_URI | SNAPSHOT | DISK")
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: This should be updated:
"Incorrect '--source' usage: --source {SNAPSHOT | DISK} | --source VHD_BLOB_URI [--source-storage-account-id ID]") 

Also the error on line 863 should reference the same usage error string.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py (hunk 1 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: The value varies per image, so there is no fixed value list.
I did reword a bit to make it not like a flag though.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Because of server limitations, for cross subscriptions scenario, it is hard to sniff out the account resource id from the blob uri

Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Right.  So, do you have to have access rights to that subscription to create a disk from it?  Does it matter if the blob is public or private?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py (hunk 1 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: How would someone be expected to know what values can be used?

Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: This is a good question. Till today, there is only one fixed value of "Windows_Server", but when CRP team rolls out the client OS image in near future, we will get a new license type "Windows_Client". I think for foreseeable future we can just use a choice list

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Correct. It does not matter the blob is public or private, the owner of the subscription need to give you the access to the account through RBAC. Of course, this doesn't make lots of sense for a public blob, but that is not the main scenario

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py (hunk 4 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: I will adjust it.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: So the storage account ID is used by the backend, not directly in the command?

Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Correct
